<html lang='en'>
  <head>
    <meta content='width=device-width, initial-scale=1.0' name='viewport' />
    <link href='bootstrap/css/bootstrap.css' rel='stylesheet' />
    <link href='style.css' rel='stylesheet' />

  </head>
  <body>
    <div class='navbar navbar-inverse navbar-static-top' role='navigation'>
      <div class='container'>
        <div class='navbar-header'>
          <button class='navbar-toggle' data-toggle='collapse' type='button' data-target='.navbar-collapse'>
            <span class='sr-only'>Toggle navigation</span>
            <span class='icon-bar'></span>
            <span class='icon-bar'></span>
            <span class='icon-bar'></span>
          </button>
          <a class='navbar-brand' href='./'>peptagram</a>
        </div>
        <div class='navbar-collapse collapse'>
          <ul class='nav navbar-nav'>
            <li>
              <a href='./'>home</a>
            </li>
            <li>
              <a href='vizhelp.html'>viz help</a>
            </li>
            <li class='active'>
              <a href='vizgen.html'>viz generation</a>
            </li>
            <li>
              <a href='api.html'>programming api</a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div id='banner'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-12'>
            <h1>PEPTAGRAM</h1>
            <h4>visual comparison of proteomics experiments</h4>

          </div>
        </div>
      </div>
    </div>
    <div id='tintedtext'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-8'>

<h1>Generating Visualizations</h1>
<p>The purpose of <code>peptagram</code> is to visually compare peptide hits in proteins across different proteomics experiments.</p>
<p><code>Peptagram</code> makes visualizations of peptide hits of results generated from Morpheus, Mascot, Maxqaunt, ProteinProphet/PeptideProphet, X!Tandem and ProteinPilot.</p>
<p>Some test data is provided in <code>example_data.zip</code>. If you unzip this in the <code>peptagram</code> directory, you should get a sub-directory <code>example_data</code>, which contains a list of sub-directories <code>mascot</code>, <code>morpheus</code> etc.</p>
<p>For each search engine, there is a separate program to generate a peptagram. Let's look at one in detail in the next section.</p>          </div>
        </div>
      </div>
    </div>
    <div id='plaintext'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-8'>

<h2>Example: Making Morpheus Peptagrams</h2>
<p>Morpheus is a fast search engine designed for high-quality data. As Morpheus does not come with a bundled viewer, <code>peptagram</code> provides a unique tool to view Morpheus search results.</p>
<p>The scripts that process Morpheus have <code>morpheus_peptagram</code> in their name:</p>
<ul>
<li><code>mac_morpheus_peptagram.command</code> - which can be clicked in Finder on Mac OSX</li>
<li><code>win_morpheus_peptagram.bat</code> - which can be clicked in WIndows Filer Explorer</li>
<li><code>do_morpheus_peptagram.py</code> - which is run on the command-line as <code>python do_morpheus_peptagram.py -i</code></li>
</ul>
<h3>Test</h3>
<p>To run an automated test, unzip <code>example_data.zip</code>. It should create an <code>example_data</code> directory containing sub-directories such as <code>morpheus</code>, <code>mascot</code> etc. Then to run the test:</p>
<pre><code>python do_morpheus_peptagram.py test
</code></pre>
<h3>Create a Peptagram with the GUI</h3>
<p>To use <code>morpheus_peptagram</code>, first start the program, then you should get a window that looks something like this:</p>
<p><img alt="" src="morpheus.png" /></p>
<p>To load the Morpheus files, click <code>+ PSMs.tsv</code> files and select all the .PSMs.tsv files that you want to compare. <code>morpheus_peptagram</code> will figure out the corresponding .protein_group.tsv files from these filenames.</p>
<p>Once selected, you'll see a list of files:</p>
<p><img alt="" src="files.png" /></p>
<p>You can now reorder the .PSMs.tsv files into your preferred order by dragging the &#9776; icon.</p>
<p>Then you can scroll down to the bottom, and click the submit button:</p>
<p><img alt="" src="submit.png" /></p>
<p>If there are any errors encountered, they'll appear below the submit button. Hopefully the error message at the last line will help you trouble-shoot the problem, and then you can click submit again.</p>
<p><img alt="" src="error.png" /></p>
<p>If it worked, you'll get a link to a newly created directory containing your peptagram:</p>
<p><img alt="" src="success.png" /></p>
<p>Later, you might want to tweak the options, such as loading the spectra from the original .mzML files, or restricting the display to matches by the Q-score.</p>
<p><img alt="" src="options.png" /></p>          </div>
        </div>
      </div>
    </div>
    <div id='tintedtext'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-8'>

<h2>Available Peptagram builders</h2>
<ul>
<li><code>morpheus_peptagram</code> - processes Morpheus search engine results. Requires the modifications.tsv file for modified peptides, and optionally the .mzML file if you want to display spectra</li>
<li><code>mascot_peptagram</code> - processes Mascot .dat. Shows the spectra for each PSM. Requires the original .fasta file to get the full length protein sequences</li>
<li><code>maxquant_peptagram</code> - process Maxquant txt/summary directories. Shows the matched ions in the spectra only. Requires the original fasta file to get the full length protein sequences.</li>
<li><code>prophet_peptagram</code> - processes the TPP's prot.xml and pep.xml files. Requires the original .fasta file to get the full length protein sequences</li>
<li><code>xtandem_peptagram</code> - processes X!Tandem search results. Shows the spectra for each PSM.</li>
<li><code>pilot_peptagram</code> - processes Protein Pilot .txt/.csv result files. Requires the original .fasta file to get the full length protein sequences</li>
</ul>          </div>
        </div>
      </div>
    </div>
    <div id='plaintext'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-8'>

<h2>Managing the size of the peptagrams</h2>
<p>As some peptagrams can get really big easily, there are a number of options to filter out low-quality matches, from using quality scores like pep and ionscore.</p>
<p>Some of the programs group similar proteins together (Morpheus, Maxquant, Protein Prophet). This effectively reduces the number of proteins, and therefore the size of the resultant peptagram. For the other search engines, you can load a text-file of seqids of contaminants to exclude. Or you can load another text-file of seqids that will be shown.</p>
<p>As well, you can limit the display to tryptic, semi-tryptic or modified peptides.</p>          </div>
        </div>
      </div>
    </div>
    <div id='tintedtext'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-8'>

<h2>Combining Different Peptagrams</h2>
<p>After you're created the peptagrams using the scripts described above, you can now recombine and edit them using <code>reorder_peptagram</code>.</p>
<p>With <code>reorder_peptagram</code>, you can load existing peptagrams, and stack all the rows of the different peptagrams against each other.</p>
<p>Then you reorder and relabel each row, before saving it to a new peptagram.</p>
<p>The one catch is that the sequence IDs of your proteins have to match. One way is to ensure that you use the same FASTA database for your search and that the sequence IDs are single words separated by a space from the description.</p>          </div>
        </div>
      </div>
    </div>
    <div id='footer'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-8'>
<p><code>peptagram</code> was developed at the <a href="http://monash.edu/proteomics">Monash Proteomics Facility</a>.  <br>
Source code hosted at <a href="http://github.com/boscoh/peptagram">http://github.com/boscoh/peptagram</a> under BSD license.   <br>
&copy; 2013. Bosco K. Ho.</p>          </div>
        </div>
      </div>
    </div>
    <script src='jquery-1.10.2.min.js'></script>
    <script src='bootstrap/js/bootstrap.js'></script>
  </body>
</html>