<html lang='en'>
  <head>
    <meta content='width=device-width, initial-scale=1.0' name='viewport' />
    <link href='bootstrap/css/bootstrap.css' rel='stylesheet' />
    <link href='style.css' rel='stylesheet' />

  </head>
  <body>
    <div class='navbar navbar-inverse navbar-static-top' role='navigation'>
      <div class='container'>
        <div class='navbar-header'>
          <button class='navbar-toggle' data-toggle='collapse' type='button' data-target='.navbar-collapse'>
            <span class='sr-only'>Toggle navigation</span>
            <span class='icon-bar'></span>
            <span class='icon-bar'></span>
            <span class='icon-bar'></span>
          </button>
          <a class='navbar-brand' href='./'>peptagram</a>
        </div>
        <div class='navbar-collapse collapse'>
          <ul class='nav navbar-nav'>
            <li>
              <a href='./'>home</a>
            </li>
            <li>
              <a href='vizhelp.html'>viz help</a>
            </li>
            <li class='active'>
              <a href='vizgen.html'>viz generation</a>
            </li>
            <li>
              <a href='api.html'>programming api</a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div id='banner'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-12'>
            <h1>PEPTAGRAM</h1>
            <h4>hassle-free visualisation of proteomics data</h4>

          </div>
        </div>
      </div>
    </div>
    <div id='tintedtext'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-8'>

<h1>Generating Visualizations</h1>
<p>The purpose of <code>peptagram</code> is to visually compare peptide hits in proteins between experiments.</p>
<p><code>Peptagram</code> can read peptide-spectrum match results from Morpheus, Mascot, Maxqaunt, ProtXML/PepXML, X!Tandem and ProteinPilot.</p>
<p>For each type of search results, we provide a script that can be run on the command-line, or in a GUI. For a set of example test data, please download <a href="http://monash.edu/proteomics/peptagram/example_data.zip">http://monash.edu/proteomics/peptagram/example_data.zip</a> into your <code>peptagram</code> directory, and unzip into sub-directory called <code>example_data</code>.</p>
<p>Let's look at generating a peptagram from the Morpheus search results.</p>          </div>
        </div>
      </div>
    </div>
    <div id='plaintext'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-8'>

<h3>Example: making Peptagrams from Morpheus search results</h3>
<p>Morpheus is a fast search engine designed for high-quality data. As Morpheus does not come with a bundled viewer, <code>peptagram</code> provides a unique tool to view Morpheus search results.</p>
<p>The scripts that process Morpheus all start with <code>morpheus_peptagram</code>. They are:</p>
<ul>
<li><code>morpheus_peptagram.command</code> - which can be clicked in Finder on Mac OSX</li>
<li><code>morpheus_peptagram.bat</code> - which can be clicked in WIndows Filer Explorer</li>
<li><code>morpheus_peptagram.py</code> - which is run on the command-line as <code>python morpheus_peptagram.py -i</code></li>
</ul>
<p>If you've downloaded the example data, you can run a default test:</p>
<pre><code>python morpheus_peptagram.py test
</code></pre>
<p>To use <code>morpheus_peptagram</code>, first start the program, then you should get a window that looks something like this:</p>
<p><img alt="" src="morpheus.png" /></p>
<p>To load the Morpheus files, click <code>+ PSMs.tsv</code> files and select all the .PSMs.tsv files that you want to compare. <code>morpheus_peptagram</code> will figure out the corresponding .protein_group.tsv files from these filenames.</p>
<p>Once selected, you'll see a list of files:</p>
<p><img alt="" src="files.png" /></p>
<p>You can now reorder the .PSMs.tsv files into your preferred order by dragging the &#9776; icon.</p>
<p>Then you can scroll down to the bottom, and click the submit button:</p>
<p><img alt="" src="submit.png" /></p>
<p>If there are any errors encountered, they'll appear below the submit button. Hopefully the error message at the last line will help you trouble-shoot the problem, and then you can click submit again.</p>
<p><img alt="" src="error.png" /></p>
<p>If it worked, you'll get a link to a newly created directory containing your peptagram:</p>
<p><img alt="" src="success.png" /></p>
<p>Later, you might want to tweak the options, such as loading the spectra from the original .mzML files, or restricting the display to matches by the Q-score.</p>
<p><img alt="" src="options.png" /></p>          </div>
        </div>
      </div>
    </div>
    <div id='tintedtext'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-8'>

<h3>Available Peptagram builders</h3>
<ul>
<li><code>morpheus_peptagram</code> - processes Morpheus search engine results. Requires the modifications.tsv file for modified peptides, and optionally the .mzML file if you want to display spectra</li>
<li><code>mascot_peptagram</code> - processes Mascot .dat. Shows the spectra for each PSM. Requires the original .fasta file to get the full length protein sequences</li>
<li><code>maxquant_peptagram</code> - process Maxquant txt/summary directories. Shows the matched ions in the spectra only. Requires the original fasta file to get the full length protein sequences.</li>
<li><code>prophet_peptagram</code> - processes the TPP's prot.xml and pep.xml files. Requires the original .fasta file to get the full length protein sequences</li>
<li><code>xtandem_peptagram</code> - processes X!Tandem search results. Shows the spectra for each PSM.</li>
<li><code>pilot_peptagram</code> - processes Protein Pilot .txt/.csv result files. Requires the original .fasta file to get the full length protein sequences</li>
</ul>
<p>As some peptagrams can get really big easily, there are a number of options to filter out low-quality matches, from using quality scores like pep and ionscore. As well, you can always load text-files of contaminants to exclude, and/or seqids of interest. As well, there are a number of peptide filtering options.</p>          </div>
        </div>
      </div>
    </div>
    <div id='tintedtext'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-8'>

<h3>Reordering and Combining Multiple Peptagrams</h3>
<p>After you're created the peptagrams using the scripts described above, you can now recombine and edit them using:</p>
<ul>
<li><code>reorder_peptgram</code></li>
</ul>
<p>With <code>reorder_peptgram.py</code>, you can load existing peptagrams, stacking all the individual rows of the peptagrams against each other.</p>
<p>Then you reorder and relabel each row to your hearts content, before saving it to a new peptagram.</p>
<p>The only catch is that the sequence IDs of your proteins have to match exactly. You must ensure that the sequence ID used in the corresponding FASTA databases are the same.</p>          </div>
        </div>
      </div>
    </div>
    <div id='footer'>
      <div class='container'>
        <div class='row'>
          <div class='col-lg-8'>
<p><code>peptagram</code> was developed at the <a href="http://monash.edu/proteomics">Monash Proteomics Facility</a>.  <br>
Source code hosted at <a href="http://github.com/boscoh/peptagram">http://github.com/boscoh/peptagram</a> under BSD license.   <br>
&copy; 2013. Bosco K. Ho.</p>          </div>
        </div>
      </div>
    </div>
    <script src='jquery-1.10.2.min.js'></script>
    <script src='bootstrap/js/bootstrap.js'></script>
  </body>
</html>